wandb/sdk/launch/agent/agent.py

```diff
@@ -546,7 +551,17 @@ def _check_run_finished(self, job_tracker: JobAndRunStatus) -> bool:
         known_error = False
         try:
             run = job_tracker.run
-            status = run.get_status().state
+            try:
+                status = run.get_status().state
+            except ApiException:
+                # Pod not reachable
+                wandb.termlog(f"{LOG_PREFIX}Requeueing job: {launch_spec}")
+                launch_add(
+                    config=launch_spec,
```

Comment:
Does launch_spec already have a `run_id` set?  In order to make sure the re-queued run continues the current one we'll need to make sure we set that. Also if we don't want users to have to set the `resume` arg in their call to `wandb.init` we'll want to also set a key in the launch spec, so we can control the `WANDB_RESUME` envirnoment variable